### PR TITLE
Rename @ECLASS-VARIABLE to @ECLASS_VARIABLE

### PR DESCRIPTION
**Pull request by ulm**

### CODE REVIEW

File: src/pkgcore/ebuild/eclass.py [hunk 1]
```diff
@@ -235,6 +235,29 @@ def _supported_eapis(self, block, tag, lineno):
...
+            '@PRE_INHERIT': ('pre_inherit', False, self._tag_bool, False),
+            '@USER_VARIABLE': ('user_variable', False, self._tag_bool, False),
+            '@OUTPUT_VARIABLE': ('output_variable', False, self._tag_bool, False),
+            '@DESCRIPTION:': ('description', True, self._tag_multiline_str, None),
+        }
+        super().__init__(tags)
+
+
+# For backwards compatibility, can be removed after 2022-12-31
+class EclassVarBlockCompat(ParseEclassDoc):
```
Comment (mgorny) [inline src/pkgcore/ebuild/eclass.py:260]: Hmm, no chance you could use `EclassVarBlock` as the base class and just override `tag`?

Comment (ulm) [reply src/pkgcore/ebuild/eclass.py:260]: That's why I had asked in #-qa. :smile:
```
<@mgorny> i think duplicating is fine
<@mgorny> or even better
```

`@ECLASS-VARIABLE` is also used in `tags`, so would have to override that too (how?).

Comment (ulm) [reply src/pkgcore/ebuild/eclass.py:260]: Sorry, but I fear that my python skills aren't sufficient to produce a more elegant solution. Especially, not sure how the child class would access `tags`, which is defined inside the `__init__` method of its parent.

File: tests/ebuild/test_eclass.py [hunk 2]
```diff
@@ -64,13 +64,13 @@ def make_eclass(name, provides=None):
...
 
-# @ECLASS-VARIABLE: _FOO_INTERNAL_ECLASS_VAR
+# @ECLASS_VARIABLE: _FOO_INTERNAL_ECLASS_VAR
 # @INTERNAL
 # @DEFAULT_UNSET
 # @DESCRIPTION:
 # Internal variable.
 
-# @ECLASS-VARIABLE: FOO_PUBLIC_ECLASS_VAR
+# @ECLASS_VARIABLE: FOO_PUBLIC_ECLASS_VAR
```
Comment (mgorny) [inline tests/ebuild/test_eclass.py:73]: How about keeping the old syntax in one of them for better test coverage? Either that or adding a dedicated test for the compat logic.

**Comment by thesamesam** [comment]
conceptual ack

File: tests/ebuild/test_eclass.py (hunk 2 above)
Comment (ulm) [reply tests/ebuild/test_eclass.py:73]: I've added `FOO_ANOTHER_ECLASS_VAR` with the old syntax.

**Review by mgorny — APPROVED** [review]
Thanks for doing this! Maybe I'll experiment with trying to simplify the code later.

**Pull request closed without merging**